r/tools/nixlibs.R

```diff
@@ -814,8 +816,14 @@ set_thirdparty_urls <- function(env_var_list) {
   env_var_list
 }
 
-is_feature_requested <- function(env_varname, default = env_is("LIBARROW_MINIMAL", "false")) {
-  env_value <- tolower(Sys.getenv(env_varname))
+is_feature_requested <- function(env_varname, env_var_list, default = env_is("LIBARROW_MINIMAL", "false")) {
+  # look in our env_var_list first, if it's not found there go to
+  # the actual environment
+  env_value <- tolower(env_var_list[[env_varname]])
+  if (is.null(env_value)) {
+    env_value <- tolower(Sys.getenv(env_varname))
+  }
```

Comment:
This is a change in behavior that warrants some thought. Though I _will_ say this is how I thought this worked already — or rather I thought `is_feature_requested()` was looking at `env_var_list` and not *only* looking at the literal environment. It's possible this was a mistake / it should have never looked at the environment but instead be looking at the `env_var_list`. But I kept it for now in case this is actually intended.

Reply:
The original intent of the function was to check whether the user requested a feature. `env_var_list` doesn't contain any user-requested features, at least not in the active sense. But now that you're putting some default selections in there, it makes sense to check it here. The function is currently only used to check for S3/GCS, to know whether we need to have openssl and curl.

Reply:
AAAAH I was missing the `user-` part there, that makes it make more sense now.

Reply:
See, this is why passive voice is bad.